lib/messages/commands/inv.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+var Message = require('../message');
+var inherits = require('util').inherits;
+var bitcore = require('bitcore');
+var utils = require('../utils');
+var BufferReader = bitcore.encoding.BufferReader;
+var BufferWriter = bitcore.encoding.BufferWriter;
+var Inventory = require('../../inventory');
+var _ = bitcore.deps._;
+
+var magicNumber = bitcore.Networks.defaultNetwork.networkMagic.readUInt32LE(0);
+
+/**
+ * @param {Object|Array} - options - If options is an array will use as "inventory"
+ * @param {Array} options.inventory - An array of inventory items
+ * @extends Message
+ * @constructor
+ */
+function InvMessage(options) {
+  if (!(this instanceof InvMessage)) {
+    return new InvMessage(options);
+  }
+  if(!options) {
+    options = {};
+  }
+  Message.call(this, options);
+  this.command = 'inv';
+  this.magicNumber = magicNumber;
+
+  var inventory;
+  if (_.isArray(options)) {
+    inventory = options;
+  } else {
+    inventory = options.inventory;
+  }
+  utils.checkInventory(inventory);
+  this.inventory = inventory;
+}
+inherits(InvMessage, Message);
+
+/**
+ * @param {Buffer|String} hash - The hash of the transaction inventory item
+ * @returns {InvMessage}
+ */
+InvMessage.forTransaction = function(hash) {
+  return new InvMessage([Inventory.forTransaction(hash)]);
+};
+
+/**
+ * @param {Buffer|String} hash - The hash of the block inventory item
+ * @returns {InvMessage}
+ */
+InvMessage.forBlock = function(hash) {
+  return new InvMessage([Inventory.forBlock(hash)]);
+};
+
+/**
+ * @param {Buffer|String} hash - The hash of the filtered block inventory item
+ * @returns {InvMessage}
+ */
+InvMessage.forFilteredBlock = function(hash) {
+  return new InvMessage([Inventory.forFilteredBlock(hash)]);
+};
```

Comment:
I don't like that these 3 methods (`forBlock`, `forFilteredBlock` and `forTransaction`) are repeated in some classes. Why don't you use a generic method builder like: https://github.com/bitpay/bitcore-p2p/blob/master/lib/messages.js#L974 ?